expression/builtin_math_vec.go

```diff
@@ -697,11 +697,38 @@ func (b *builtinCeilIntToDecSig) vecEvalDecimal(input *chunk.Chunk, result *chun
 }
 
 func (b *builtinTruncateIntSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinTruncateIntSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+
+	if err := b.args[1].VecEvalInt(b.ctx, input, buf); err != nil {
+		return err
+	}
+	result.MergeNulls(buf)
+	i64s := result.Int64s()
+
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			i64s[i] = 0
+			continue
+		}
+		if buf.GetInt64(i) < 0 {
```

Comment:
Please use `buf.Int64s()` instead of using `buf.GetInt64(i)` and then we can reduce some function calls.

Reply:
OK, I have fixed it

Reply:
Nice, and please also update the benchmark results.